src/mongocxx/test/index_view.cpp

```diff
@@ -172,6 +172,53 @@ TEST_CASE("create_one", "[index_view]") {
         REQUIRE_NOTHROW(indexes.create_one(keys1.view(), options.view()));
         REQUIRE_THROWS_AS(indexes.create_one(keys2.view(), options.view()), operation_exception);
     }
+
+    SECTION("commitQuorum option") {
+        collection coll = db["index_view_create_one_commit_quorum"];
+        coll.drop();
+        coll.insert_one({});  // Ensure that the collection exists.
+        index_view indexes = coll.indexes();
+
+        auto key = make_document(kvp("a", 1));
+        index_model model(key.view());
+        options::index_view options;
+
+        auto commit_quorum_regex =
+            Catch::Matches("(.*)commit( )?quorum(.*)", Catch::CaseSensitive::No);
+
+        using namespace test_util;
+        bool is_supported =
+            get_max_wire_version(mongodb_client) >= 9 && get_topology(mongodb_client) != "single";
+        CAPTURE(is_supported);
+
+        SECTION("works with int") {
+            options.commit_quorum(1);
+            if (is_supported) {
+                REQUIRE_NOTHROW(indexes.create_one(model, options));
+            } else {
+                REQUIRE_THROWS_WITH(indexes.create_one(model, options), commit_quorum_regex);
+            }
+        }
+
+        SECTION("fails with invalid int") {
+            options.commit_quorum(-1);
+            REQUIRE_THROWS_WITH(indexes.create_one(model, options), commit_quorum_regex);
+        }
+
+        SECTION("works with string") {
```

Comment:
This section, and the section below, appear to be the ones failing.

As the capturing of `is_supported` shows, `is_supported` is false, an an exception is expected.

IIUC an exception is expected because the test runs against a single topology, but "majority" should only be supported on a replica set. And the section below expects a server to always reject the value "bad_str"

I still believe this is due to the intermediate 4.3.0 server version this is running against on zSeries. The server may have implemented validation after 4.3.0, so the reported wire version is still 9, but invalid commitQuorum values are still accepted. https://jira.mongodb.org/browse/SERVER-41846 in particular (which has fixVersion 4.3.1) seems related. 

We could investigate further and check with the server team. But IMO, it is less important that we test server side validation of these values, and more important that we test that `commitQuroum` works when supported, and that the driver returns a client side error if the wire version is < 9.

I'd argue for removing the "fails with invalid string" test below, and only testing the "is_supported" case here.

Reply:
Per our discussion, I skipped single typologies and got rid of the `invalid with...` tests.